org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -3173,6 +3243,18 @@ protected void saveNew_afterSetID()
 	public String get_WhereClause (boolean withValues)
 	{
 		StringBuilder sb = new StringBuilder();
+
+		if (!Util.isEmpty(m_uuid, true))
+		{
+			sb.append(getUUIDColumnName()).append("=");
+			if (withValues)
+				sb.append(m_uuid);
+			else
+				sb.append("?");
+
+			return sb.toString();
+		}
+
```

Comment:
suggestion: rename this to get_whereclause(string uuid, boolean withValues). add get_whereclause(boolean withvalues) {return get_whereclause(null, withvalues);}

Reply:
In refactored code, this is handled.